src/blueprints/tv2/factories/tv2-transition-factory.ts

```diff
@@ -0,0 +1,110 @@
+import { Action, MutateActionMethods, PieceAction } from '../../../model/entities/action'
+import { ActionType } from '../../../model/enums/action-type'
+import { Piece, PieceInterface } from '../../../model/entities/piece'
+import { PieceType } from '../../../model/enums/piece-type'
+import { TransitionType } from '../../../model/enums/transition-type'
+import { PieceLifespan } from '../../../model/enums/piece-lifespan'
+import { Tv2AtemLayer, Tv2SourceLayer } from '../value-objects/tv2-layers'
+import { AtemMeTimelineObject, AtemTransition, AtemType } from '../../timeline-state-resolver-types/atem-types'
+import { DeviceType } from '../../../model/enums/device-type'
+import { Tv2BlueprintTimelineObject } from '../value-objects/tv2-metadata'
+import { TimelineObject } from '../../../model/entities/timeline-object'
+
+const FRAME_RATE: number = 25
+
+const FRAMES_FOR_MIX_TRANSITION: number = 25 // TODO: Get from configuration.
+
+const MIX_TRANSITION_ID: string = 'nextTakeHasMixTransitionAction'
+
+export class Tv2TransitionFactory {
+
+  public isTransitionAction(action: Action): boolean {
+    return [MIX_TRANSITION_ID].includes(action.id)
+  }
+
+  public getMutateActionMethods(action: Action): MutateActionMethods | undefined {
+    switch (action.id) {
+      case MIX_TRANSITION_ID: {
+        return {
+          updateActionWithPlannedPieceData: (action: Action, plannedPiece: Piece) => this.updateAtemMeInput(action, plannedPiece),
+          plannedPiecePredicate: (piece: Piece) => piece.timelineObjects.some(timelineObject => timelineObject.layer === Tv2AtemLayer.PROGRAM)
+        }
+      }
+    }
+  }
+
+  public createMixTransitionAction(): PieceAction {
+    const pieceInterface: PieceInterface = {
+      id: 'mixTransitionActionPiece',
+      name: 'Mix transition',
+      partId: '',
+      type: PieceType.TRANSITION,
+      layer: Tv2SourceLayer.JINGLE,
+      pieceLifespan: PieceLifespan.WITHIN_PART,
+      transitionType: TransitionType.IN_TRANSITION,
+      isPlanned: false,
+      start: 0,
+      duration: this.getTimeFromFrames(FRAMES_FOR_MIX_TRANSITION),
+      postRollDuration: 0,
+      preRollDuration: 0,
+      tags: [],
+      timelineObjects: []
+    }
+
+    return {
+      id: MIX_TRANSITION_ID,
+      name: 'Mix transition on next take',
+      type: ActionType.INSERT_PIECE_AS_NEXT,
+      data: pieceInterface
+    }
+  }
+
+  private getTimeFromFrames(frames: number): number {
```

Comment:
As long as the unit is specifed ones during some part of a calculation, it should be expected that the unit doesn't change until otherwise specifed, or easily deducable from the calculations done. Where the line goes for what is easily deducable, and what is not, is the hard part.

A good example of this, is when you are calculation duration. As long as `InMs` is specified in one of the used input values, the result of the calculation is expected to also be `InMs`, without stating so in the name, unless the name specifically states that it is now e.g `InSeconds`. 
Sometimes the change of unit can be deduced easily, with the calculation being done, e.g dividing a value known to be `InMs` by 1000, means now it is in seconds. Because of this, specifying the unit can becomes less important, but will still useful, if further calculation is done with it.

Reply:
Personally, I think it would be nice if the variable name gave an indication of if the value is in milliseconds, seconds, minutes, hours etc. So something like `InMs`, `secondsToWaitForX` etc

I think `epoch` should be used when it's important that the `number` provided is an epoch timestamp and there is no other indications for it. For example the `CallbackScheduler` interface:
```
export interface CallbackScheduler {
  start(epochTimeToExecuteCallback: number, callback: () => void): void
  stop(): void
}
```

I think having a "standard" unit might not be a good option. Everything is case by case dependent and it's easy to forget when we are deviating from the standard. 
But if we are going for some sort of standard, then I suggest to use milliseconds :)

Reply:
I am also for having the units. It makes it less confusing. But is it something that we want to introduce on the entities (when we have the freedom to do so)? E.g. `Piece.start` becomes `Piece.startOffsetInMs` or something like that?

Reply:
You asks some good questions @KvelaGorrrrnio 

Something in me wants to let it just stay `Piece.start`, but that kinda goes against my previous comment and I don't have any good arguments to let it stay `Piece.start`